scripts/operatorhub-deploy.sh

```diff
@@ -0,0 +1,173 @@
+EOT
+}
+
+if [[ "${1:-}" == --send-pr ]]; then
+    send_pr=true
+    shift
+elif [[ "${1:-}" =~ ^-.*$ ]]; then
+    cat <<EOT
+
+error: You issued operatorhub-deploy.sh $@
+Did you want to set the --send-pr flag?
+
+EOT
+    usage
+    exit 1
+fi
+
+if ! [[ "${1:-}" =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+    echo -e "\nerror: VERSION argument not passed, or in wrong format\n" >&2
+    usage
+    exit 1
+fi
+new_version="$1"
+eunomia_root=$(
+    cd "$(dirname "$0")/.."
+    pwd
+)
+
+# "operator-sdk olm-catalog gen-csv" command requires a specific project layout:
+# https://github.com/operator-framework/operator-sdk/blob/v0.8.1/doc/user/olm-catalog/generating-a-csv.md#configuration
+echo -e "${WHITE}Creating directory structure to satisfy operator-sdk"
+echo -e "\n- Generating operator.yaml and role.yaml${NC}"
+# Create temporary directory to store intermediate helm-generated files
+tmp_dir=$(mktemp -d)
+helm template "${eunomia_root}/deploy/helm/eunomia-operator/" \
+    --set eunomia.operator.deployment.operatorHub=true \
+    --set eunomia.operator.image.tag="v${new_version}" \
+    --output-dir "$tmp_dir"
+# Create operator.yaml and role.yaml
+cat "${tmp_dir}/eunomia-operator/templates/deployment.yaml" \
+    >"${eunomia_root}/deploy/operator.yaml"
+cat "${tmp_dir}/eunomia-operator/templates/clusterrole-operators.yaml" \
+    "${tmp_dir}/eunomia-operator/templates/role-operator.yaml" \
+    >"${eunomia_root}/deploy/role.yaml"
+# Delete temporary directory
+rm -rf "$tmp_dir"
+
+# Get previous versions from OperatorHub
+echo -e "\n${WHITE}- Downloading previous releases of Eunomia from OperatorHub${NC}"
+tmp_dir=$(mktemp -d)
+
+git -C "$tmp_dir" clone https://github.com/operator-framework/community-operators
+git_context="git -C ${tmp_dir}/community-operators"
+hub_context="hub -C ${tmp_dir}/community-operators"
+
+mkdir -p "${eunomia_root}/deploy/olm-catalog"
+cp -R -f "${tmp_dir}/community-operators/upstream-community-operators/eunomia" \
+    "${eunomia_root}/deploy/olm-catalog"
+echo -e "${WHITE}Files successfully generated${NC}"
+
+# Find latest version
+latest=$(grep currentCSV "${eunomia_root}/deploy/olm-catalog/eunomia/eunomia.package.yaml" | awk -F"eunomia.v" '{print $2}')
+
+# Generate ClusterServiceVersion (CSV) of a new version
+echo -e "\n${WHITE}Generating a new ClusterServiceVersion yaml file${NC}"
+(
+    cd "$eunomia_root" &&
+        operator-sdk olm-catalog gen-csv --csv-version "$new_version" --from-version "$latest"
+)
+echo -e "\n${WHITE}- Clean up temp files${NC}"
+rm -f "${eunomia_root}/deploy/operator.yaml" "${eunomia_root}/deploy/role.yaml"
+
+# By default copy gitopsconfig crd from latest version
+cp "${eunomia_root}/deploy/olm-catalog/eunomia/${latest}/gitopsconfigs.eunomia.kohls.io.crd.yaml" \
+    "${eunomia_root}/deploy/olm-catalog/eunomia/${new_version}/"
+
+# Make currentCSV in eunomia.package.yaml point to the new version
+# Note: using -i.bkup in order for it to work both on Mac and Linux
```

Comment:
[optional] the stackoverflow link that was on next line in an earlier version of this PR could still be helpful; I personally liked it and didn't think it should be removed